libraries/SDFS/src/SDFS.cpp

```diff
@@ -0,0 +1,155 @@
+/*
+ SDFS.cpp - file system wrapper for SdFat
+ Copyright (c) 2019 Earle F. Philhower, III. All rights reserved.
+
+ Based on spiffs_api.cpp which is:
+ | Copyright (c) 2015 Ivan Grokhotkov. All rights reserved.
+
+ This code was influenced by NodeMCU and Sming libraries, and first version of
+ Arduino wrapper written by Hristo Gochkov.
+
+ This file is part of the esp8266 core for Arduino environment.
+
+ This library is free software; you can redistribute it and/or
+ modify it under the terms of the GNU Lesser General Public
+ License as published by the Free Software Foundation; either
+ version 2.1 of the License, or (at your option) any later version.
+
+ This library is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General Public
+ License along with this library; if not, write to the Free Software
+ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+#include "SDFS.h"
+#include "SDFSFormatter.h"
+#include <FS.h>
+
+using namespace fs;
+
+
+#if !defined(NO_GLOBAL_INSTANCES) && !defined(NO_GLOBAL_SDFS)
+static sdfs::SDFSImpl *_SDFSImpl = new sdfs::SDFSImpl();
```

Comment:
Why save this pointer? It doesn't seem to serve any purpose.
How about something like this:
```
FS SDFS = FS(FSImplPtr(new sdfs::SDFSImpl()));
```
?

Reply:
No idea, but it's cleaned up in the latest push.